File: exercises/ddd_infrastructure_layer/java/src/main/java/infrastructure/InMemoryApplicationRepository.java
package infrastructure;

import java.util.Map;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

import domain.application.Application;
import domain.application.ApplicationId;
import domain.application.ApplicationRepository;

public final class InMemoryApplicationRepository implements ApplicationRepository {
    private final Map<UUID, Application> entries = new ConcurrentHashMap<>();

    @Override
    public Application getById(ApplicationId id) {
        return entries.get(id.getValue());
    }

    @Override
    public void save(Application application) {
        entries.put(application.getId().getValue(), application);
    }
}

File: exercises/ddd_infrastructure_layer/typescript/src/infrastructure/InMemoryApplicationRepository.ts
import type { ApplicationRepository } from '../domain/application/ApplicationRepository';
import type { Application } from '../domain/application/Application';
import type { ApplicationId } from '../domain/application/ApplicationId';

export class InMemoryApplicationRepository implements ApplicationRepository {
  private entries = new Map<string, Application>();

  async getById(id: ApplicationId) {
    return this.entries.get(id.value);
  }

  async save(application: Application) {
    this.entries.set(application.id.value, application);
  }
}

